server/chat-plugins/lottery.js

```diff
@@ -0,0 +1,213 @@
+'use strict';
+
+const LOTTERY_FILE = 'config/chat-plugins/lottery.json';
+
+/** @type {typeof import('../../lib/fs').FS} */
+const FS = require(/** @type {any} */('../../.lib-dist/fs')).FS;
+
+/** @type {{[roomid: string]: {name: string, markup: string, participants: {[ip: string]: string}}}} */
+let lotteries = {};
+try {
+	lotteries = require(`../../${LOTTERY_FILE}`);
+} catch (e) {
+	if (e.code !== 'MODULE_NOT_FOUND' && e.code !== 'ENOENT') throw e;
+}
+if (!lotteries || typeof lotteries !== 'object') lotteries = {};
+
+class Lottery {
+	/**
+     * @param {string} roomid
+     */
+	constructor(roomid) {
+		this.roomid = roomid;
+	}
+	/**
+	 * @param {string} name
+	 * @param {string} markup
+	 */
+	create(name, markup) {
+		if (!lotteries[this.roomid]) {
+			lotteries[this.roomid] = {name, markup, participants: {}};
```

Comment:
It doesn't make a whole lot of sense to both have a `Lottery` class and keep state it depends on outside of it. `lotteries` should be a map of roomids to `Lottery` instances and `Lottery` itself should be keeping track of `name`, `markup`, and `participants` instead.

Also, same deal with not using `{}` here for holding on to participants.

Reply:
I'd also make it so `lotteries` is only used from commands instead, since it's just the commands that really need to care about this state.

Reply:
Refactored to low-level functions that manipulate lotteries.